staging/src/k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go

```diff
@@ -94,6 +94,10 @@ type NodeResourcesFitArgs struct {
 	// A resource group name can't contain '/'.
 	// +listType=atomic
 	IgnoredResourceGroups []string `json:"ignoredResourceGroups,omitempty"`
+
+	// ScoringStrategy selects the node resource scoring strategy.
+	// +listType=atomic
+	ScoringStrategy *ScoringStrategy `json:"scoringStrategy,omitempty"`
```

Comment:
We should not add this field in v1beta1 so that we don't have to worry about conflicts with existing plugins. If users want the new behavior, they can simply use v1beta2.

Reply:
we will need to explicitly list it as "deprecated" in v1beta1, it may help users migrate to the new plugin having it in v1beta1.

Reply:
ok, lets remove from v1beta1 and list the plugin as deprecated in v1beta1 to prevent listing it in the pluginset.

Reply:
It's fine to allow it. The plugin would just return immediately because there is no strategy set. I don't think it's worth the custom validation.

Reply:
it would be confusing to users, we allow it but does nothing.

Reply:
Then just allowing both plugins should be fine. Worst case, there are 2 scores calculating the same, essentially a weight of 2. They don't really conflict.

Reply:
the conflict validation is not complex...

Reply:
The current implementation is suggesting possibility of expansion. We don't want to encourage that.

Reply:
Expansion of what?

Reply:
Of conflict checking. In general, we don't want to check for conflicts on released API versions.

Reply:
We are not checking for conflicts against released plugins, this is a new plugin and we are adding validation to that specific newly released plugin, there is no change of behavior. Also, any misinterpretation of intent can be fixed with a comment if we are worried about future additions to this list (which we will remove with v1beta1 deprecation).

Reply:
> Then just allowing both plugins should be fine. Worst case, there are 2 scores calculating the same, essentially a weight of 2. They don't really conflict.

This adds even more confusion to the users.

What problem are you trying to solve? having those few lines of code is not significant compared to protecting users from surprises.

Reply:
I'm trying to prevent a misuse of the current conflict detection machinery that could break old APIs in the future.

Reply:
But that applies to all validation logic, we can break old APIs by altering any existing validation logic, and we can only prevent that with reviews. As for the new conflict checking logic, as I mentioned, we can clarify intent with proper commenting that discourages expending it, and it only exists to support a legacy case. In my opinion, protecting users from surprises takes higher priority here.

Reply:
> this is a new plugin

The plugin was in 1.21:

https://github.com/kubernetes/kubernetes/blob/release-1.21/staging/src/k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go#L79-L93

I don't think we should add new capabilities to the v1beta1 config and simultaneously add conflict detection on that API version. It's easier to leave v1beta1 alone, limit the scheduler behavior changes to fields which can only be set using v1beta2 config, and limit deprecations to new config versions

Reply:
This is a new **scoring** plugin, the new validation on conflict is not a breaking change.